azimuth/config.py

```diff
@@ -308,15 +372,45 @@ class SyntaxConfig(CommonFieldsConfig):
     syntax: SyntaxOptions = SyntaxOptions()
 
 
+class LanguageConfig(CommonFieldsConfig):
+    # Language configuration sets multiple config values; see reference dictionary/ies for details
+    # TODO Correct grammar when code decision has been made
+    language: SupportedLanguage = SupportedLanguage.en
+
+
 class AzimuthConfig(
     PerturbationTestingConfig,
     SimilarityConfig,
     DatasetWarningConfig,
     SyntaxConfig,
+    LanguageConfig,
     extra=Extra.forbid,
 ):
+    # TODO @fred But _why_ should this class remain empty? :) Should it just be empty of data
+    #  attributes, and are methods ok?
     # This class should remain empty!
-    pass
+    # pass
+    @root_validator()
+    def dynamic_language_config_values(cls, values):
+        if values["behavioral_testing"]:
+            values["behavioral_testing"].neutral_token.prefix_list = (
+                values["behavioral_testing"].neutral_token.prefix_list
+                or prefix_list_per_language[values["language"]]
+            )
+            values["behavioral_testing"].neutral_token.suffix_list = (
+                values["behavioral_testing"].neutral_token.suffix_list
+                or suffix_list_per_language[values["language"]]
+            )
+        values["syntax"].subj_tags = (
+            values["syntax"].subj_tags or subj_tags_per_language[values["language"]]
+        )
+        values["syntax"].obj_tags = (
+            values["syntax"].obj_tags or obj_tags_per_language[values["language"]]
+        )
+        values["syntax"].spacy_model = (
+            values["syntax"].spacy_model or spacy_model_per_language[values["language"]]
+        )
```

Comment:
I got an idea that would define the defaulting logic closer to where the fields are defined:
- Create a method in `SyntaxOptions`:
```py
    def set_defaults(self, language: SupportedLanguage):
        # To see all dep tag options for a spacy model: spacy.load(SPACY_MODEL).get_pipe("parser").labels
        if language == SupportedLanguage.en:
            self.subj_tags = self.subj_tags or ["nsubj", "nsubjpass"]
            self.obj_tags = self.obj_tags or ["dobj", "pobj", "obj"]
            self.obj_tags = self.obj_tags or "en_core_web_sm"
        elif language == SupportedLanguage.fr:
            self.subj_tags = self.subj_tags or ["nsubj", "nsubjpass"]
            self.obj_tags = self.obj_tags or ["obj", "iobj", "obl:arg", "obl:agent", "obl:mod"]
            self.obj_tags = self.obj_tags or "fr_core_news_md"
```
- And call it here:
```suggestion
        values["syntax"].set_defaults(values["language"])
```
- With that approach, I think we can get rid of the constant dictionaries that we were not sure where and how to define.

What do you think?

Reply:
Let me check that I understand what you're suggesting? So, the first part of `SyntaxOptions` would still be the same, where we set things to empty lists or strings, and the user can provide their own value(s). And we still have the `root_validator` doing the dynamic selection, but we're just replacing code that checks an external dictionary with code that runs an internal method? 

I'm assuming that by the time the root_validator is run, the empty list/string is considered falsy because we're not doing `pre=True`, so that part of the logic still works? I.e. the logic is written in the parent class, but run in the child class, so it's ok?

I like the idea of defining a `set_defaults` method in `SyntaxOptions` so that the logic is closer to where the defaults are set, as you said. My only question is about making it easy to see all the things that are changed with a different language. Take a look at the commented code and question at the top of the file - I'll tag you - and let me know what you think in the context of that suggestion? In theory, all the language-specific values could still be set in a (MEGA) dictionary like that, but we could still have a `set_defaults` method to make it more clear what's happening when you're reading the code.